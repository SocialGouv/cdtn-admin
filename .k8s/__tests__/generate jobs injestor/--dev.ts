//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate jobs/injestor --dev ", async () => {
  expect(
    await getEnvManifests("dev", "jobs/injestor", {
      ...project("cdtn-admin").dev,
      CDTN_ADMIN_ENDPOINT: "https://cdtn-admin.fabrique.social.gouv.fr/api/graphql",
      ES_INDEX_PREFIX: `cdtn-dev`,
      KUBE_NAMESPACE: "foo-XYZ-master-dev2",
      RANCHER_PROJECT_ID: "c-bar:p-foo",
    })
  ).toMatchSnapshot();
});
