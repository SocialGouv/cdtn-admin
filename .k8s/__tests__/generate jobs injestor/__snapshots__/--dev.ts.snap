// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate jobs/injestor --dev  1`] = `
"---
apiVersion: v1
kind: ConfigMap
metadata:
  name: injester
data:
  NODE_ENV: production
  CDTN_ADMIN_ENDPOINT: http://hasura-cdtn-admin/v1/graphql
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
  name: injester
  namespace: null
spec:
  encryptedData:
    ELASTICSEARCH_URL: >-
      AgBOS+1BrmLEGQoU4WQL+RPs9c6+LGyLrLf0M3GrQ7g86EDP1CXX61jcHBGeE7x9rKz3S0YmyDi7RHvzYcMsJypLrHd6E3wgkmrq44TYgn8HkMMsMHCWh24afaPR1zHrkCGAqhD37897JbpMOuNqToMo0bhHsMhrcy51KTweOcnkF+gEVhbCisF1nVdYbj6ql6CIcIc0XMx89ICaW7aK4UrRb9d/ZxZfOOv7qqsNIVzTUsx/TKMvuvmgarZS8pcrgiS4LISynlnvM1pdd7UFAXqE+ki3JxfcmWiasBtCFQm3rmOwm1+RYJqLqO/mPHQMp8I0DQoinZ8EC5m5zH290vzK7EQnpZxU02EWEoLiAVsQLp5mE2HtB9tM9znNSyF52p/lWQZ291Nyse5pCGIdDLc8v4Y6XQjAnGMDZBoZod7ikRfRIU1rhUE4PhP9IoCk4Ap9FxPFPUu3ezbq/XlTx4Qjaqrx5D5ZrRiOHDbaoyrLy/YocEK4Su66js86zsuJPn6J8xhAYyi+funAqeRf0DTUUvDhEpTUC8mjSPoyydQ4jlA6Tg+b/FSMlRvDIOHdnNIVu6L/DCcE2bRrpvbemBL8QE/EkRMAO8GXe+5IlmUle2FeqBQHGOydH/qcPS37IJfuUB50IEHthkQIjzL4uxVi8ptz4tXCROq2vgMhBCq9471x0JLJcYYVWbY8tNmFdoJQKrv4vFPVJBdbYP6ojnnSeHP6IOWkak2izo7eCdYvjwbGJnHjA51XqrV9AcETacYWqXcn0yc6LXRZFMWv1vp4Mg+/JTFXA4I585uGh8t0+VQogg==
    ELASTICSEARCH_DATA_TOKEN: >-
      AgBUJtRy5N7iEcLhEuYzuXU3V5x/1IcfJ+j1nD7Am36swyljTKoiGbe0rTlKl8+N9E1pITT1RNWhLBGPEhNlT1aBgYIdZ6+SIbzoyd7aVcck591M6DDVPPTAAKPi3JZlJCAfkr/nKBaDNhf++fD49D7kITGo+if60oEDKj+i7FvpyH4i3C8G48ktX/IW2H31mvPDXqSR37jkOrIxn/3T7l57plmh67kLGkHKIQdGbp4qxw2uV/Sk0wYxCpQXQzJwE2HGVKRMvE0lPRs+g634FHx7+3TrQ4iBtreQJbxfuRKz+TgenSy1z0VJvoC5hEn7+XvpqG82G5j/XTZJ0vs1NFcTQs/vYweNToXIklfDCDtiaalLM3XwGutZxHvTAwaOf1h6SCmKHZLjMjrirLMdsvTK7KW5OLwHcgG2Izdp9t0WEgNb5XngfNTJmy86ttlafJtNkezNanyjWTu+zIbo5icf55DfHOu2T5/9/FKqaMLsrW+I3c/eS/IUM4tXFNoyeA9ihMb/LUAHveoSQt8ne7dzUdGWzCI6mJsU2saa+PKO2ctonBMXGQq0DEb6d+xBbSlLkD0EPN8BIvL0qM5LSQU5jtHrni1uij5T9oFeL4aJGMIDTDtG/YH7j5kRceDSDoDpg2yv2GZ8HOducFXVD/UUW8DYfiw1NVr/5cWbXa7nXHhb4NXmALkh5oU6wx3xHFW/XptFuCvNGppcRqkrBHYDrCpJnZGC1XKrnFBVh5+DB9UT3nKcxuAwj9nxv1v/PYwSTE7epfbzxdsudqg=
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
      name: injester
    type: Opaque
---
apiVersion: batch/v1
kind: Job
metadata:
  name: injestor-es-424242
  namespace: cdtn-admin-24-branch-42
spec:
  backoffLimit: 0
  template:
    spec:
      initContainers:
        - args:
            - http://hasura-cdtn-admin/healthz
          image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/docker/wait-for-http:4.5.4
          imagePullPolicy: Always
          name: wait-for-wait-for-hasura
          resources:
            limits:
              cpu: 20m
              memory: 32Mi
            requests:
              cpu: 5m
              memory: 16Mi
      containers:
        - name: ingester-es
          image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/cdtn-admin/ingester-es:8843083edb7f873cad1d1420731a60773594ffae
          imagePullPolicy: IfNotPresent
          resources:
            limits:
              cpu: '1'
              memory: 1Gi
            requests:
              cpu: 500m
              memory: 512Mi
          envFrom:
            - configMapRef:
                name: injester
            - secretRef:
                name: injester
          env:
            - name: ES_INDEX_PREFIX
              value: cdtn-dev-cdtn-admin-branch
      restartPolicy: Never
  ttlSecondsAfterFinished: 86400
"
`;
